### PR TITLE
Error correction on getting-started.rst page.

### PR DESCRIPTION
**Pull request by jwbaumann**
I commented "import: warnings" on line 112. That line throws an error on my fresh install, as follows:

% cabal build myfirstapp
Errors encountered when parsing cabal file ./myfirstapp.cabal:

myfirstapp.cabal:29:3: error:
Undefined common stanza imported: warnings

   28 | executable myfirstapp 
   29 |   import: warnings
      |   ^

% 

That line is not present in the docs for 3.6. I'll leave it for someone to suss out how it got added. It should probably be removed.

---
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.
I'm new, so I don't actually know for a fact that this change conforms to the above. It's a pretty simple errata which is likely terribly confusing to newbies.

Please also shortly describe how you tested your change. Bonus points for added tests!
I removed the problematic line from my own code and the myfirstapp project was successfully built when it was removed.

### CODE REVIEW

**Comment by Martinsos** [comment]
Hey @jwbaumann , thanks for the PR!

I added this line while I was recently updating the getting-started.rst to correctly reflect how the latest version of cabal 3.7.0.0 works.

Basically, what I did was follow the instructions from the getting-started.rst in order to create a new cabal project, and then I looked up the generated myfirstapp.cabal and copied the contents of `executable myfirstapp` stanza into the getting-started.rst, to show where the new dependency needs to be added.

This is what myfirstapp.cabal looks for me, as generated by the version of cabal from the current master branch:

<details>
<summary>myfirstapp.cabal</summary>

```
cabal-version:      3.0
-- The cabal-version field refers to the version of the .cabal specification,
-- and can be different from the cabal-install (the tool) version and the
-- Cabal (the library) version you are using. As such, the Cabal (the library)
-- version used must be equal or greater than the version stated in this field.
-- Starting from the specification version 2.2, the cabal-version field must be
-- the first thing in the cabal file.

-- Initial package description 'myfirstapp' generated by
-- 'cabal init'. For further documentation, see:
--   http://haskell.org/cabal/users-guide/
-- 
-- The name of the package.
name:               myfirstapp

-- The package version.
-- See the Haskell package versioning policy (PVP) for standards
-- guiding when and how versions should be incremented.
-- https://pvp.haskell.org
-- PVP summary:     +-+------- breaking API changes
--                  | | +----- non-breaking API additions
--                  | | | +--- code changes with no API change
version:            0.1.0.0

-- A short (one-line) description of the package.
-- synopsis:

-- A longer description of the package.
-- description:

-- The license under which the package is released.
license:            NONE

-- The package author(s).
author:             sosic.martin@gmail.com

-- An email address to which users can send suggestions, bug reports, and patches.
maintainer:         Martin Sosic

-- A copyright notice.
-- copyright:
build-type:         Simple

-- Extra doc files to be distributed with the package, such as a CHANGELOG or a README.
extra-doc-files:    CHANGELOG.md

-- Extra source files to be distributed with the package, such as examples, or a tutorial module.
-- extra-source-files:

common warnings
    ghc-options: -Wall


executable myfirstapp
    -- Import common warning flags.
    import:           warnings

    -- .hs or .lhs file containing the Main module.
    main-is:          Main.hs

    -- Modules included in this executable, other than Main.
    -- other-modules:

    -- LANGUAGE extensions used by modules in this package.
    -- other-extensions:

    -- Other library packages from which modules are imported.
    build-depends:    base ^>=4.14.3.0

    -- Directories containing source files.
    hs-source-dirs:   app

    -- Base language which the package is written in.
    default-language: Haskell2010
```

</details>

As you can see, `executable myfirstapp` contains `import: warnings` stanza, which is basically importing the `common warnings` stanza that is couple of lines above it. And, this works well for me, no complaints from cabal. Here is a bit in current docs about `common` and `import`: https://cabal.readthedocs.io/en/latest/cabal-package.html?highlight=import#common-stanzas .

Only thing I could think of being a cause of a problem here is if for some reason on my machine different myfirstapp.cabal is being generated than it normally should be, possibly due to some misconfig, but that sounds a bit far-fetched to me - we could try checking it though.

What is your myfirstapp.cabal looking like @jwbaumann , could you paste it here? Also, which version of cabal are you using (run `cabal --version`)?

@Mikolaj , on a somewhat separate note, do you think it is useful to correctly replicate the state of `executable myfirstapp` stanza in the `myfirstapp.cabal` in getting-started.rst, as I have done, or would it be better to ommit some of the advanced/optional lines and leave just the most important (build-depends, main-is, hs-source-dirs)? First one makes stronger connection between the docs and what is happening on user's machine, while the second option might be a bit less fragile.

**Review by Martinsos — CHANGES_REQUESTED** [review]
Asked a question, let's wait for an answer before we decide on the next steps!

**Comment by Mikolaj** [comment]
@Martinsos: hi! regarding your question: lets ask the user(s) ;)

**Comment by Mikolaj** [comment]
@jwbaumann: what do you think? In the absence of strong preferences we'd go with our gut feeling which, I guess, which is to show the user new stuff, elicit questions and then answer them. Do you think an explanation in the text would be helpful, about how the `import` comes from `init` invocation and will be described in this and that section?

**Comment by Mikolaj** [comment]
@jwbaumann: ping?

**Comment by Mikolaj** [comment]
@jwbaumann: when you some time to revisit this, please let me know and I will re-open. For now I'm closing the PR. Thank you all!

**Comment by jwbaumann** [comment]
Sorry, the replies above got lost in my copious emails.

My cabal version in 3.6.2.0, even after running "cabal update". This is on MacOS 10.15.7 Catalina.

Being new to cabal and Haskell, while still having extensive programming experience with PHP, Javascript, and other non-compiled languages, I would go for non-fragile and informative approach in tutorial code. in other words, I would do something like this:

-- the following is part of a fully up-to-date modern standard canonical implementation / installation of cabal
-- however, we have reports of this line causing build failures on some systems
-- we suggest first trying "cabal build myfirstapp" with this line commented out
-- if the build is successful, try again with the line uncommented
-- to explore what might be happening to cause the error, visit { insert appropriate link or reference here }
-- here is the troublesome line (note that the uncommented line should have 2 leading spaces)
--   import: warnings

The full text of my "myfirstapp.cabal" file is

~~~
cabal-version:      3.6
name:               myfirstapp
version:            0.1.0.0

-- A short (one-line) description of the package.
-- synopsis: Just Learning Here

-- A longer description of the package.
-- description: Still just learning

-- A URL where users can report bugs.
-- bug-reports:

-- The license under which the package is released.
-- license:
author:             Jeffrey W Baumann
maintainer:         jwbaumann@linkedresources.com

-- A copyright notice.
-- copyright: 2022
-- category: open source
extra-source-files: CHANGELOG.md

common firststeps
  build-depends: base ^>=4.14.3.0, haskell-say ^>=1.0.0.0
  

executable myfirstapp 
  import: warnings
  main-is: Main.hs

-- Modules included in this executable, other than Main.
-- other-modules:

-- LANGUAGE extensions used by modules in this package.
-- other-extensions:
  build-depends: base ^>=4.14.3.0, haskell-say ^>=1.0.0.0
  hs-source-dirs: app
  default-language: Haskell2010
~~~

From my perspective as a learner, I do not want unexpected errors (there are plenty enough already), and I DO want the ability to understand what is causing the error. In this particular case, why is this error occurring? Where is cabal looking for that "warnings" module to import, and why is it not being found despite having run "cabal update"? Why does "cabal update" not update me to version 3.7.0?

I am not fully familiar with the process/etiquette for "pull requests" or other debugging dialogs on GitHub, which adds to a general feeling of opaqueness in this environment.

**Comment by jwbaumann** [comment]
Mikolaj Konarski,

Sorry for the delay. I just posted this comment at

https://github.com/haskell/cabal/pull/7895

-----------------------------------------------------------
Sorry, the replies above got lost in my copious emails.

My cabal version in 3.6.2.0, even after running "cabal update". This is on MacOS 10.15.7 Catalina.

Being new to cabal and Haskell, while still having extensive programming experience with PHP, Javascript, and other non-compiled languages, I would go for non-fragile and informative approach in tutorial code. in other words, I would do something like this:

-- the following is part of a fully up-to-date modern standard canonical implementation / installation of cabal
-- however, we have reports of this line causing build failures on some systems
-- we suggest first trying "cabal build myfirstapp" with this line commented out
-- if the build is successful, try again with the line uncommented
-- to explore what might be happening to cause the error, visit { insert appropriate link or reference here }
-- here is the troublesome line (note that the uncommented line should have 2 leading spaces)
-- import: warnings

The full text of my "myfirstapp.cabal" file is

cabal-version:      3.6
name:               myfirstapp
version:            0.1.0.0

-- A short (one-line) description of the package.
-- synopsis: Just Learning Here

-- A longer description of the package.
-- description: Still just learning

-- A URL where users can report bugs.
-- bug-reports:

-- The license under which the package is released.
-- license:
author:             Jeffrey W Baumann
maintainer:         ***@***.***

-- A copyright notice.
-- copyright: 2022
-- category: open source
extra-source-files: CHANGELOG.md

common firststeps
  build-depends: base ^>=4.14.3.0, haskell-say ^>=1.0.0.0
  

executable myfirstapp 
  import: warnings
  main-is: Main.hs

-- Modules included in this executable, other than Main.
-- other-modules:

-- LANGUAGE extensions used by modules in this package.
-- other-extensions:
  build-depends: base ^>=4.14.3.0, haskell-say ^>=1.0.0.0
  hs-source-dirs: app
  default-language: Haskell2010
From my perspective as a learner, I do not want unexpected errors (there are plenty enough already), and I DO want the ability to understand what is causing the error. In this particular case, why is this error occurring? Where is cabal looking for that "warnings" module to import, and why is it not being found despite having run "cabal update"? Why does "cabal update" not update me to version 3.7.0?

I am not fully familiar with the process/etiquette for "pull requests" or other debugging dialogs on GitHub, which adds to a general feeling of opaqueness in this environment.
-----------------------------------------------------------------

Thanks for your attempts to address this. I have been doing the "Plutus Pioneers Program (ADA, Cardano, IOHK, Plutus, blockchain. Smart contracts) and it is stretching my brain and my time to it's limits. I blame my newness to Haskell. And cabal. And GitHub. And nix.

Jeff Baumann

**Comment by Mikolaj** [comment]
Hi @jwbaumann, 

Thank you for responding. I'm sorry about your experience with Haskell/cabal/cabal bugtracker. 

Let's try to start afresh. To clarify how you got there, could you tell where is your myfirstapp.cabal file coming from? E.g., a link?

You have the newest stable cabal, don't worry. I assumed, since you are proposing to change the unstable branch, that you are able to test the unstable branch (3.7), which involves cloning and compiling. But we we can do that together, so no need to do anything.

**Comment by gbaz** [comment]
i think its the same issue as https://github.com/haskell/cabal/issues/8014 -- just "getting started" being out of sync with the cabal a user is using.

**Pull request closed without merging**
